src/utils/cartUtils.ts

```diff
@@ -150,6 +160,10 @@ export async function finishCheckout(
         }
         if (transactionRecord.id && Address.id) {
             for (const item of cartItems) {
+                const product = await db.Products.findOne({ where: { id: item.productId }, transaction });
+                if (!product || product.quantity < 1) {
+                    throw new CustomError(`${product?.title} is out of stock or You ordered more than what is in stock.`, 400);
```

Comment:
Not sure you need to throw an error in this case - If you order a item that is out of stock - the request is still valid and the response could be just a string explaining the case.